packages/main/src/StepInput.ts

```diff
@@ -367,7 +367,15 @@ class StepInput extends UI5Element implements IFormElement {
 	}
 
 	get _valuePrecisioned() {
-		return this.value.toFixed(this.valuePrecision);
+		if (this.value === 0 || !this.value.toString().includes(".") || (this.value.toString().split(".")[1].length <= this.valuePrecision)) {
```

Comment:
> Should we round the value when we have an "Error" state?

No, we should not round the value based on the `valueState`. The `valueState` is set to notify the end user that the entered value is invalid. Also I dont think using `valueState` as a condition is a good idea, as as we set the valueState, it could be easily prevented by the app developer. Meaning for us the value could be invalid, but the `valueState` could still be `valueState.None` or something else.